python/pyarrow/tests/test_compute.py

```diff
@@ -1635,6 +1636,23 @@ def test_is_null():
     assert result.equals(expected)
 
 
+def test_is_nan():
+    arr = pa.array([1, 2, 3, None, np.nan])
+    result = arr.is_nan()
+    expected = pa.array([False, False, False, None, True])
+    assert result.equals(expected)
+
+    with pytest.raises(
+            ArrowNotImplementedError, match="has no kernel matching input types"):
+        arr = pa.array(["1", "2", None], type=pa.string())
```

Comment:
```suggestion
    arr = pa.array(["1", "2", None], type=pa.string())
    with pytest.raises(
            ArrowNotImplementedError, match="has no kernel matching input types"):
```

And the same for the one below (although unlikely here, but that avoids that it might be accidentally the array creation that is erroring, instead of calling is_nan)